### PR TITLE
RemoteSha and package name

### PR DESCRIPTION
**Pull request by wlandau**
This PR has two important updates:

1. Use RemoteSha when comparing versions, thanks to help from @jeroen in https://github.com/r-universe-org/help/issues/377. 
2. Change the package name to `r.releases.utils` to avoid confusion with https://github.com/r-releases/help/issues/6.

### CODE REVIEW

**Comment by wlandau** [comment]
Just fixed the merge conflicts from merging #10

File: R/record_versions.R [hunk 1]
```diff
@@ -41,15 +37,25 @@ record_versions <- function(
...
+  repo = "https://r-releases.r-universe.dev"
 ) {
-  out <- utils::available.packages(repos = repos)
-  out <- out[, c("Package", "Version", "MD5sum")]
-  out <- as.data.frame(out)
+  listing <- paste(
+    contrib.url(repos = repo, type = "source"),
+    "PACKAGES.json?fields=RemoteSha",
+    sep = "/"
+  )
```
Comment (shikokuchuo) [inline R/record_versions.R:50]: can use your neat `file.path()` trick here

File: R/record_versions.R [hunk 2]
```diff
@@ -41,15 +37,25 @@ record_versions <- function(
...
+  out <- jsonlite::stream_in(
+    con = url(listing),
```
Comment (shikokuchuo) [inline R/record_versions.R:52]: When I've used `jsonlite::stream_in()` in the past, I've wrapped `url()` in `gzcon()`, so it'd be `gzcon(url(listing))` here. Theoretically the time saved by reducing transmission bytes dwarfs any extra processing for the compression.

File: R/record_versions.R [hunk 3]
```diff
@@ -32,7 +28,7 @@ record_versions <- function(
   previous <- read_versions_previous(manifest = manifest)
   new <- update_version_manifest(current = current, previous = previous)
   jsonlite::write_json(x = new, path = manifest, pretty = TRUE)
-  aligned <- versions_aligned(manifest = new, check_hash = check_hash)
+  aligned <- versions_aligned(manifest = new)
```
Comment (shikokuchuo) [inline R/record_versions.R:31]: Given `versions_aligned()` is simplified with only one argument, I would find it easier to follow if it was just inlined here.

**Review by shikokuchuo — CHANGES_REQUESTED** [review]

File: R/record_versions.R (hunk 1 above)
Comment (wlandau-lilly) [reply R/record_versions.R]: done in 8cd45a33ce3624496fce57ff479bcdd60fa28116

File: R/record_versions.R (hunk 2 above)
Comment (wlandau-lilly) [reply R/record_versions.R]: Nice! Done in 3655a8db9eb0013c5aad80f5f0fb1b5eb17cad1a

**Comment by wlandau** [comment]
Thanks for approving! I will go ahead and make a release, change the name of this repo, and update the automation to use the correct package name etc.

**Pull request merged**